examples/llm_inference/android/app/src/main/java/com/example/mediapipe/llminference/InferenceModel.kt

```diff
@@ -0,0 +1,60 @@
+package com.example.mediapipe.llminference
+
+import android.content.Context
+import com.google.mediapipe.tasks.core.Delegate
+import com.google.mediapipe.tasks.genai.llminference.LlmInference
+import java.io.File
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharedFlow
+
+class InferenceModel private constructor(context: Context) {
+    private var llmInference: LlmInference
+
+    private val modelExists: Boolean
+        get() = File(MODEL_PATH).exists()
+
+    private val _partialResults = MutableSharedFlow<Pair<String, Boolean>>(
+        extraBufferCapacity = 1,
+        onBufferOverflow = BufferOverflow.DROP_OLDEST
+    )
+    val partialResults: SharedFlow<Pair<String, Boolean>> = _partialResults
+
+    init {
+        if (!modelExists) {
+            throw IllegalArgumentException("Model not found at path: $MODEL_PATH")
+        }
+
+        val options = LlmInference.LlmInferenceOptions.builder()
+            .setModelPath(MODEL_PATH)
+            .setDelegate(Delegate.GPU)
+            .setNumDecodeStepsPerSync(3)
+            .setMaxSequenceLength(1024)
+            .setTopK(1)
+            .setRandomSeed(0)
+            .setTemperature(0f)
```

Comment:
Can you remove these to use the defaults (which are slightly different)? Applies to topK, randomSeed, temperature

Reply:
Done